met-api/src/met_api/resources/language.py

```diff
@@ -75,43 +76,82 @@ def delete(language_id):
             return str(err), HTTPStatus.NOT_FOUND
 
 
-@cors_preflight('GET, POST, OPTIONS, PATCH, DELETE')
-@API.route('/')
-class Languages(Resource):
-    """Resource for managing multiple languages."""
+@cors_preflight('GET, OPTIONS')
+@API.route('/tenant/<int:tenant_id>')
+class TenantLanguages(Resource):
+    """Resource for getting existing language-tenant mappings."""
 
     @staticmethod
     @cross_origin(origins=allowedorigins())
-    def get():
-        """Fetch list of languages."""
+    def get(tenant_id):
+        """Fetch list of languages associated with a given tenant ID."""
         try:
-            languages = LanguageService.get_languages()
+            languages = LanguageService.get_languages_by_tenant(tenant_id)
             return (
                 jsonify(LanguageSchema(many=True).dump(languages)),
                 HTTPStatus.OK,
             )
         except (KeyError, ValueError) as err:
             return str(err), HTTPStatus.INTERNAL_SERVER_ERROR
 
+
+@cors_preflight('POST, OPTIONS')
+@API.route('/<int:language_id>/tenant/<int:tenant_id>')
+class ManageLanguageMappings(Resource):
+    """Resource for adding or removing language-tenant relationships."""
+
+    @staticmethod
+    def post(language_id, tenant_id):
+        """Create language-tenant mapping."""
+        try:
+            return LanguageService.map_language_to_tenant(language_id, tenant_id)
+        except IntegrityError as e:
+            # Catching language code already exists error
+            detail = (
+                str(e.orig).split('DETAIL: ')[1]
+                if 'DETAIL: ' in str(e.orig)
+                else 'Duplicate entry.'
+            )
+            raise BusinessException(
+                str(detail), HTTPStatus.INTERNAL_SERVER_ERROR
+            ) from e
+
+
+@cors_preflight('DELETE, OPTIONS')
+@API.route('/mappings/<int:language_mapping_id>')
+class DeleteLanguageMapping(Resource):
+    """Resourse for removing language-tenant relationships."""
+
+    @staticmethod
+    # @_jwt.requires_auth
```

Comment:
I realize this is temporary, but do remember to add appropriate authentication to your routes before merging to main

Reply:
I made a note about it in the PR description but yes for sure, I'll have to circle back to auth once I push changes for the frontend. It'll be easier to test it when I'm making requests from the browser